mailpoet/assets/js/src/automation/editor/index.tsx

```diff
@@ -69,6 +103,10 @@ function Editor(): JSX.Element {
     setShowActivatePanel(!showActivatePanel);
   };
 
+  if (isBooting) {
+    updateingActiveWorkflowNotPossible();
+    setIsBooting(false);
+  }
```

Comment:
I think that instead of using state here it would be more idiomatic to use `useEffect` with an empty dependency array (or perhaps a dependency array that won't change). Again I'm not sure if this might have any kind of negative effect and I know this is just a temporary solution (which seems to be working as intended locally for me), so I'll once again defer to @JanJakes.

Reply:
@johnolek That's a good catch. Yes, it should be in `useEffect`, the dependency will be `isBooting`, I guess, and the linters may force you to add `updateingActiveWorkflowNotPossible` as well, which is fine, since it's a module-global, which will always have the same identity.

@websupporter Even if this works, it's fragile and sometimes React can throw errors that the state was updated during a render (not in an effect).

I also noticed `updateing` -> should be `updating` but probably not important in a "temporary" code.

Btw. this is one of the things I really dislike about React. They don't invent a new "language" like Svelte, but they heavily bend JS itself and force us to use unintuitive constructs. In SolidJS, this code would be perfectly valid. Not only valid, it would actually be the correct way to do it.

Reply:
Thank you both for your input.

useEffect: [ae1d2da](https://github.com/mailpoet/mailpoet/pull/4462/commits/ae1d2da4203a880cd7c5f153ee04150b4b42a015)

Typo: [6119835](https://github.com/mailpoet/mailpoet/pull/4462/commits/611983506af2ed0d9261c77c5711b754912591a6)